tests/functional/adapter/custom_schema_tests/test_custom_database.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import os
+from dbt.tests.util import (
+    check_relations_equal,
+    check_table_does_exist,
+    run_dbt
+)
+from tests.functional.adapter.custom_schema_tests.seeds import seed_queries
+
+
+_VIEW_1_SQL = """
+select * from {{ target.schema }}.seed
+""".lstrip()
+
+_VIEW_2_SQL = """
+{{ config(database='alt') }}
+select * from {{ ref('view_1') }}
+""".lstrip()
+
+_VIEW_3_SQL = """
+{{ config(database='alt', materialized='table') }}
+
+
+with v1 as (
+
+    select * from {{ ref('view_1') }}
+
+),
+
+v2 as (
+
+    select * from {{ ref('view_2') }}
+
+),
+
+combined as (
+
+    select last_name from v1
+    union all
+    select last_name from v2
+
+)
+
+select
+    last_name,
+    count(*) as count
+
+from combined
+group by 1
+""".lstrip()
+
+_CUSTOM_DB_SQL = """
```

Comment:
This doesn't appear to be used anywhere, unless there are some built-in keywords in dbt that I don't recognize.

Reply:
```
@pytest.fixture(scope="class")
    def macros(self):
        return {
            "custom_db.sql": _CUSTOM_DB_SQL,
        }
```

Reply:
Yeah, and it's  used internally to [do some automagic under the hood](https://docs.getdbt.com/docs/build/custom-databases#generate_database_name)

Reply:
@nssalian - Yeah, I saw that, but I thought it was just creating a macro that never got called.

@VersusFacit - Oh wow, look at that. So `generate_database_name()` is a "keyword". Now I know.

Reply:
@mikealfare worth checking, I removed it to make sure it was being used + @VersusFacit 's reference helps to understand this.